apps/challenges/views.py

```diff
@@ -340,3 +358,221 @@ def challenge_phase_split_list(request, challenge_pk):
+
+    # Check for test annotation file path in yaml file.
+    yaml_file_data_of_challenge_phase = yaml_file_data['challenge_phases']
+    for data in yaml_file_data_of_challenge_phase:
+        test_annotation_file = data['test_annotation_file']
+        if test_annotation_file:
+            test_annotation_file_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                         unique_folder_name,
+                                                                         extracted_folder_name,
+                                                                         test_annotation_file))
+        else:
+            response_data = {
+                'error': 'Test annotation file path for challenge phase {} does not exist'.format(data['name'])
+                }
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        if not os.path.isfile(test_annotation_file_path):
+            response_data = {
+                'error': 'Test annotation file does not exist at the \
+                         specified path in challenge phase {}'.format(data['name'])
+            }
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for challenge image in yaml file.
+    image = yaml_file_data['image']
+    if image.endswith('.jpg') or image.endswith('.jpeg') or image.endswith('.png'):
+        challenge_image_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                unique_folder_name,
+                                                                extracted_folder_name,
+                                                                image))
+        if os.path.isfile(challenge_image_path):
+            challenge_image = open(challenge_image_path, 'rb')
+            challenge_image_file = File(challenge_image)
+        else:
+            challenge_image_file = None
+    else:
+        challenge_image_file = None
+
+    try:
+        with transaction.atomic():
+            serializer = ZipChallengeSerializer(data=yaml_file_data,
+                                                context={'request': request,
+                                                         'challenge_host_team': challenge_host_team})
+            if serializer.is_valid():
+                challenge = serializer.save()
+                challenge_obj = Challenge.objects.get(pk=challenge.pk)
+                if challenge_image_file is not None:
+                    challenge_obj.image.save(image, challenge_image_file, save=True)
+                challenge_obj.evaluation_script.save(evaluation_script,
+                                                     challenge_evaluation_script_file,
+                                                     save=True)
+
+            # Create Leaderboard
+            yaml_file_data_of_leaderboard = yaml_file_data['leaderboard']
```

Comment:
Also even if there are more serializers, I think in that case we can create a serializer which simply takes list as argument and then saves it. See `ChallengeParticipantTeamListSerializer` in the code base.